MobiFlight/ExecutionManager.cs

```diff
@@ -992,6 +992,9 @@ void FsuipcCache_ConnectionLost(object sender, EventArgs e)
         void timer_Started(object sender, EventArgs e)
         {
             OnStarted?.Invoke(this, new EventArgs());
+
+            // Force all the modules awake whenver run is activated
+            mobiFlightCache.KeepConnectedModulesAwake(true);

```

Comment:
Was there a specific reason that you had decided to take the timer event?

Better options are `MobiFlightModule.Start()` method, `MobiFlightCache.Start()` method for it, or `ExecutionManager.Start()` method.

I am not even sure why are really doing this. What's the sense in it? On Start, we have an empty cache and we will always send messages down to the device if requried, e.g. the current 7-segment values.

Reply:
> On Start, we have an empty cache and we will always send messages down to the device if requried, e.g. the current 7-segment values.

That won't be sufficient to wake the devices up with the new firmware. The new firmware only resets its sleep timer when getting an unknown message or a wakeup message. That change saves us some space in the firmware.

> What's the sense in it? 

To be fair, it's not strictly necessary to have this call here. The keep awake message code in each timer tick will wake the devices up the first time the tick is called. I wasn't wild about the subtleness of that though for future people reading this code and thought it would be better to have something explicit on start that forces the wakeup so the flow is more understandable and doesn't reply on "magic" in the tick method.

> Was there a specific reason that you had decided to take the timer event?

Because the keep awake messages happen in the timer tick, so having wakeup and sleep in timer started and timer stopped puts the code together, making it easier to understand for future developers. If I move the code to module or cache it's in an entirely different place.

Reply:
Please move it at least to ExecutionManager.Start() that's the equivalent. It doesn't make sense in this callback function. The only place where the timer is enabled is in ExecutionManager.Start() - that were a future developer should find it.